app/scripts/controllers/projects.js

```diff
@@ -16,14 +16,50 @@ angular.module('openshiftConsole')
     $scope.showGetStarted = false;
     $scope.canCreate = undefined;
 
+    var sortProjects = function() {
+      // TODO: currentField is renamed in angular-patternfly 3.0
+      var sortID = _.get($scope, 'sortConfig.currentField.id', 'metadata.name');
+      var adjustedSortOrder = $scope.sortConfig.isAscending;
+
+      // Sort order must be reversed if using timestamp because the info shown to user is the relative timestamp, not the absolute
+      if (sortID === 'metadata.creationTimestamp') {
+        adjustedSortOrder = !adjustedSortOrder;
+      }
+
+      $scope.projects = _.sortByOrder($scope.projects, function(project) {
+        return _.get(project, sortID).toLowerCase();
+        //arguments[0].sortID.toLowerCase(); //project.toLowerCase();
+      }, [adjustedSortOrder ? 'asc' : 'desc']);
+      //$scope.projects = _.sortByOrder($scope.projects, [sortID], [sortOrder]);
+    };
+
+    // Set up the sort configuration for `pf-sort`.
+    $scope.sortConfig = {
+      fields: [{
+        id: 'metadata.name',
+        title: 'Name',
+        sortType: 'alpha'
+      }, {
+        id: 'metadata.annotations["openshift.io/display-name"]',
+        title: 'Display Name',
+        sortType: 'alpha'
+      }, {
+        id: 'metadata.creationTimestamp',
+        title: 'Age',
+        sortType: 'numeric'
+      }],
+      isAscending: true,
+      onSortChange: sortProjects
+    };
+
     AlertMessageService.getAlerts().forEach(function(alert) {
       $scope.alerts[alert.name] = alert.data;
     });
     AlertMessageService.clearAlerts();
 
     AuthService.withUser().then(function() {
       watches.push(DataService.watch("projects", $scope, function(projects) {
-        $scope.projects = projects.by("metadata.name");
+        $scope.projects = _.sortByOrder(projects._data, ['metadata.name'], ['asc']);
```

Comment:
You should access the `_data` property directly. It's internal. Call `projects.by("metadata.name")` before calling sort.